src/dfmt/formatter.d

```diff
@@ -281,8 +281,8 @@ private:
         {
             writeToken();
             if (index < tokens.length && (currentIs(tok!"identifier")
-                    || ( ( isBasicType(peekBack().type) || peekBackIs(tok!"identifier") ) &&
```

Comment:
I'm guessing that this sub-expression would always return true because we just wrote an identifier, and therefore `peekBackIs(tok!"identifier")` is always true. Is that the cause of the bug?

Reply:
Yes exactly.